### ISSUE TITLE
Search doesn't seem to be using current Neurovault Database

### ISSUE THREAD

**Issue by ljchang** [neurolearn/neurolearn-web#10]
I'm trying to find [this](http://neurovault.org/collections/1826/) collection on neurolearn.

@burnash:  How long does it take for the elasticnet server to index neurovault after uploading a new collection?

**Comment by ljchang** [comment]
It looks like the most recent collection is http://neurovault.org/collections/890/, which was added Oct. 9, 2015.  Could that be right?

**Comment by burnash** [comment]
You're right, 1826 is not indexed at the moment. I trigger the reindex manually from time to time, but the last update was not so long ago, definitely not a year ago. I've just reindexed search and none of the recent collections appear in the results. I think there's something wrong with the importer code. I'm looking at it now.

**Comment by burnash** [comment]
Found the age-old problem: the cache. I used it during development to make it easy on the NV API but forgot to turn it off in production. That's why new collections never appeared. I've restarted the indexer but it's choking now on some of the new images — the elastic search mapping fails to recognize several updated properties.

**Comment by ljchang** [comment]
Thanks for looking into this.  Not sure how difficult this would be, but it would be nice if the database reindexes neurovault when a new collection is added.

**Comment by burnash** [comment]
I fixed the properties issue and reindexed the search yesterday. Now it has all of the new collections. I also updated the code so it reads entire image list without 100 items limit. This means that for several large collections, select images functionality may work slow, I need to figure out how to display the list properly.

The auto-sync for the search is WIP. It turned out that adding a new collection is quite easy to do and we get it almost for free — just by periodically rerunning the indexer. The not-so-easy is removing a collection when it gets removed from NeuroVault. I used to clear the entire index and re-run the indexer, but that brings downtime so search and not desirable. I'm inclining towards keeping a list of previously indexed collections and remove missing from the index.
